vnext/Shared/Networking/WinRTHttpResource.cpp

```diff
@@ -55,6 +58,138 @@ WinRTHttpResource::WinRTHttpResource(IHttpClient &&client) noexcept : m_client{s
 
 WinRTHttpResource::WinRTHttpResource() noexcept : WinRTHttpResource(winrt::Windows::Web::Http::HttpClient{}) {}
 
+#pragma region IWinRTHttpRequestFactory
+
+IAsyncOperation<HttpRequestMessage> WinRTHttpResource::CreateRequest(
+    HttpMethod &&method,
+    Uri &&uri,
+    winrt::Windows::Foundation::Collections::IMap<winrt::hstring, IInspectable> props) noexcept /*override*/ {
+  auto request = HttpRequestMessage{std::move(method), std::move(uri)};
+  for (auto prop : props) {
+    request.Properties().Insert(prop.Key(), prop.Value());
+  }
+
+  auto iReqArgs = request.Properties().Lookup(L"RequestArgs");
+  auto reqArgs = iReqArgs.as<RequestArgs>();
+  auto self = shared_from_this();
+
+  HttpMediaTypeHeaderValue contentType{nullptr};
+  string contentEncoding;
+  string contentLength;
+
+  // Headers are generally case-insensitive
+  // https://www.ietf.org/rfc/rfc2616.txt section 4.2
+  for (auto &header : reqArgs->Headers) {
+    if (boost::iequals(header.first.c_str(), "Content-Type")) {
+      bool success = HttpMediaTypeHeaderValue::TryParse(to_hstring(header.second), contentType);
+      if (!success) {
+        if (self->m_onError) {
+          self->m_onError(reqArgs->RequestId, "Failed to parse Content-Type", false);
+        }
+        co_return nullptr;
+      }
+    } else if (boost::iequals(header.first.c_str(), "Content-Encoding")) {
+      contentEncoding = header.second;
+    } else if (boost::iequals(header.first.c_str(), "Content-Length")) {
+      contentLength = header.second;
+    } else if (boost::iequals(header.first.c_str(), "Authorization")) {
+      bool success = request.Headers().TryAppendWithoutValidation(to_hstring(header.first), to_hstring(header.second));
+      if (!success) {
```

Comment:
>success [](http://example.com/codeflow?start=11&length=7)

Just curious why it can fail as the method is called `TryAppendWithoutValidation`

Reply:
This is copied from the previous UWP module implementation.

My understanding is, some values for `Authorization` are not accepted by default by WinRT APIs, while other RN variants take them.
That method skips such strict validation.